icu4c/source/i18n/collationbuilder.cpp

```diff
@@ -1113,12 +1113,23 @@ CollationBuilder::addWithClosure(const UnicodeString &nfdPrefix, const UnicodeSt
     return ce32;
 }
 
+// ICU-22517
+// This constant define a soft limit for the addOnlyClosure to return
```

Comment:
What is a "soft" limit? It sounds like one that can sometimes be exceeded to some degree, but I don't see any special logic for that.

Maybe just drop "soft" from this comment and from the name of the constant?